inst/include/mode/mode.hpp

```diff
@@ -55,6 +55,15 @@ class container {
     }
   }
 
+  void set_state(const std::vector<double>& state) {
+    const bool individual = state.size() == n_state() * n_particles_;
+    const size_t n = individual ? 1 : n_particles_;
+    auto it = state.begin();
+    for (size_t i = 0; i < n_particles_; ++i) {
+      solver_[i].set_state(it + (i / n) * n_state());
```

Comment:
A matter of taste, and quite probably compiles to the same assembly, but this might be a bit clearer:

```suggestion
      solver_[i].set_state(it);
      if (individual) {
        it += n_state();
      }
```

(if you go with this, you can drop `n`)